portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -47,23 +47,30 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     if (userService.isUserLoggedIn()) {
-      String userEmail = userService.getCurrentUser().getEmail();
-      String logoutUrl = userService.createLogoutURL(REDIRECT_URL_AFTER_LOGOUT);
-      saveComment(request.getParameter("display-name"), request.getParameter("comment"));
+      // TODO: Use nickname instead
+      saveComment(userService.getCurrentUser().getEmail(), request.getParameter("comment"));
       response.sendRedirect("/aboutme.html");
-    } else {
-      String loginUrl = userService.createLoginURL(REDIRECT_URL_AFTER_LOGIN);
-      response.sendRedirect(loginUrl);
     }
-
   }
 
   private String convertCommentsToJson() {
     Query query = new Query("Comment");
     PreparedQuery results = datastore.prepare(query);
+
     int loggedIn = userService.isUserLoggedIn() ? 1 : 0;
+    String json = "{ \"loggedInStatus\": " + loggedIn;
 
-    String json = "{ \"loggedInStatus\": " + loggedIn +", \"comments\": [";
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      json += ", \"userEmail\": \"" + userEmail + "\"";
```

Comment:
How about displayName here instead of userEmail. Since you are not getting a Display name yet, return just the first few characters of the email as display name